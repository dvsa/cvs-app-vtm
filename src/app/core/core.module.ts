import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { GlobalErrorComponent } from './components/global-error/global-error.component';
import { SpinnerComponent } from './components/spinner/spinner.component';
import { FooterComponent } from './components/footer/footer.component';
import { HeaderComponent } from './components/header/header.component';
import { RouterModule } from '@angular/router';
import { BreadcrumbsComponent } from './components/breadcrumbs/breadcrumbs.component';
import { PageNotFoundComponent } from './components/page-not-found/page-not-found.component';
import { ServerErrorComponent } from './components/server-error/server-error.component';

@NgModule({
  declarations: [FooterComponent, HeaderComponent, GlobalErrorComponent, SpinnerComponent, BreadcrumbsComponent, PageNotFoundComponent, ServerErrorComponent],
  imports: [CommonModule, RouterModule],
  exports: [FooterComponent, HeaderComponent, GlobalErrorComponent, SpinnerComponent, BreadcrumbsComponent]
})
export class CoreModule {}
