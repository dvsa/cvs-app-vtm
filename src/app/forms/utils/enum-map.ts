import { MultiOptions } from '@forms/models/options.model';

export function getOptionsFromEnum(object: object): MultiOptions {
  return Object.values(object).map(value => ({ value, label: value.charAt(0).toUpperCase() + value.slice(1) }));
}

export function getOptionsFromEnumOneChar(object: object): MultiOptions {
  return Object.values(object).map(value => ({ value, label: value.charAt(0).toUpperCase() }));
}

export function getOptionsFromEnumAcronym(object: object): MultiOptions {
  return Object.values(object).map(value => ({ value, label: value.toUpperCase() }));
}
