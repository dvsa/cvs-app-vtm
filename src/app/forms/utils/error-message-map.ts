import { formatDate } from '@angular/common';
import { AsyncValidatorNames } from '@forms/models/async-validators.enum';
import { ValidatorNames } from '@forms/models/validators.enum';

const DEFAULT_LABEL = 'This field';
export const ErrorMessageMap: Record<string, (...args: any) => string> = {
  // Date errors
  invalidDate: (err: { error: boolean; reason: string; index: number }) => `${err.reason}`,
  invalidOption: (err: boolean, label?: string) => `${label || DEFAULT_LABEL} is invalid`,
  invalidTestResult: (err: { message: string }) => err.message,

  [ValidatorNames.AheadOfDate]: (err: { sibling: string; date: Date }, label?: string) =>
    `${label || 'This date'} must be ahead of ${err.sibling || 'the previous date'}${err.date ? formatDate(err.date, ' (dd/MM/yyyy)', 'en') : ''}`,
  [ValidatorNames.CustomPattern]: (err: { message: string }, label?: string) => `${label || DEFAULT_LABEL} ${err.message}`,
  [ValidatorNames.DateNotExceed]: (err: { sibling: string; months: number }, label?: string) =>
    `${label || 'This date'} must be less than ${err.months || 'X'} months after ${err.sibling || 'the previous date'}`,
  [ValidatorNames.Defined]: (err: boolean, label?: string) => `${label} is required`,
  [ValidatorNames.FutureDate]: (err: boolean, label?: string) => `${label || 'This date'} must be in the future`,
  [ValidatorNames.Max]: (err: { max: number }, label?: string) => `${label || DEFAULT_LABEL} must be less than or equal to ${err.max}`,
  [ValidatorNames.MaxLength]: (err: { requiredLength: number }, label?: string) =>
    `${label || DEFAULT_LABEL} must be less than or equal to ${err.requiredLength} characters`,
  [ValidatorNames.Min]: (err: { min: number }, label?: string) => `${label || DEFAULT_LABEL} must be greater than or equal to ${err.min}`,
  [ValidatorNames.MinLength]: (err: { requiredLength: number }, label?: string) =>
    `${label || DEFAULT_LABEL} must be greater than or equal to ${err.requiredLength} characters`,
  [ValidatorNames.NotZNumber]: () => "The VRM/Trailer ID cannot be in a format that is 7 digits followed by the character 'Z'",
  [ValidatorNames.PastDate]: (err: boolean, label?: string) => `${label || 'This date'} must be in the past`,
  [ValidatorNames.Pattern]: (err: boolean, label?: string) => `${label || DEFAULT_LABEL} must match a pattern`,
  [ValidatorNames.Required]: (err: boolean, label?: string) => `${label || DEFAULT_LABEL} is required`,
  [ValidatorNames.RequiredIfEquals]: (err: { sibling: string }, label?: string) => `${label || DEFAULT_LABEL} is required with ${err.sibling}`,
  [ValidatorNames.ValidateDefectNotes]: () => 'Notes is required',
  [ValidatorNames.ValidateProhibitionIssued]: () => 'Prohibition notice has not been issued.',

  [AsyncValidatorNames.RequiredIfNotAbandoned]: (err: boolean, label?: string) => `${label || DEFAULT_LABEL} is required`,
  [AsyncValidatorNames.RequiredIfNotFail]: (err: boolean, label?: string) => `${label || DEFAULT_LABEL} is required`,
  [AsyncValidatorNames.RequiredIfNotResult]: (err: boolean, label?: string) => `${label || DEFAULT_LABEL} is required`,
  [AsyncValidatorNames.RequiredIfNotResultAndSiblingEquals]: (err: boolean, label?: string) => `${label || DEFAULT_LABEL} is required`
};
