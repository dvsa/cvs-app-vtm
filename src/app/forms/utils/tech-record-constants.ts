import { FormNode } from '@forms/services/dynamic-form.types';
import { CarTechRecord } from '@forms/templates/car/car-tech-record.template';
import { ApplicantDetails } from '@forms/templates/general/applicant-details.template';
import { HgvAndTrlTypeApprovalTemplate } from '@forms/templates/general/approval-type.template';
import { Audit } from '@forms/templates/general/audit.template';
import { DocumentsTemplate } from '@forms/templates/general/documents.template';
import { HgvAndTrlBodyTemplate } from '@forms/templates/general/hgv-trl-body.template';
import { LettersTemplate } from '@forms/templates/general/letters.template';
import { ManufacturerTemplate } from '@forms/templates/general/manufacturer.template';
import { NotesTemplate } from '@forms/templates/general/notes.template';
import { PlatesTemplate } from '@forms/templates/general/plates.template';
import { HgvDimensionsTemplate } from '@forms/templates/hgv/hgv-dimensions.template';
import { HgvTechRecord } from '@forms/templates/hgv/hgv-tech-record.template';
import { tyresTemplateHgv } from '@forms/templates/hgv/hgv-tyres.template';
import { HgvWeight } from '@forms/templates/hgv/hgv-weight.template';
import { LgvTechRecord } from '@forms/templates/lgv/lgv-tech-record.template';
import { MotorcycleTechRecord } from '@forms/templates/motorcycle/motorcycle-tech-record.template';
import { PsvTypeApprovalTemplate } from '@forms/templates/psv/psv-approval-type.template';
import { PsvBodyTemplate } from '@forms/templates/psv/psv-body.template';
import { PsvBrakesTemplate } from '@forms/templates/psv/psv-brakes.template';
import { PsvDdaTemplate } from '@forms/templates/psv/psv-dda.template';
import { PsvDimensionsTemplate } from '@forms/templates/psv/psv-dimensions.template';
import { PsvNotes } from '@forms/templates/psv/psv-notes.template';
import { PsvTechRecord } from '@forms/templates/psv/psv-tech-record.template';
import { PsvTyresTemplate } from '@forms/templates/psv/psv-tyres.template';
import { PsvWeightsTemplate } from '@forms/templates/psv/psv-weight.template';
import { SmallTrailerTechRecord } from '@forms/templates/small-trailer/small-trailer-tech-record.template';

import { TechRecordReasonForCreationSection } from '@forms/templates/test-records/section-templates/techRecord_reasonForCreation/techRecordReasonForCreation.template';
import { TrlAuthIntoServiceTemplate } from '@forms/templates/trl/trl-auth-into-service.template';
import { TrlBrakesTemplate } from '@forms/templates/trl/trl-brakes.template';
import { TrlDimensionsTemplate } from '@forms/templates/trl/trl-dimensions.template';
import { TrlPurchasers } from '@forms/templates/trl/trl-purchaser.template';
import { TrlTechRecordTemplate } from '@forms/templates/trl/trl-tech-record.template';
import { tyresTemplateTrl } from '@forms/templates/trl/trl-tyres.template';
import { TrlWeight } from '@forms/templates/trl/trl-weight.template';
import { VehicleTypes } from '@models/vehicle-tech-record.model';

// The map below initializes the array of sections that the *ngFor in tech summary component's template will iterate over.
// The order in which each section is introduced in the array will determine its order on the page when rendered.
// Sections which use custom components require a FormNode object with 'name' and 'label' properties.

export const vehicleTemplateMap = new Map<VehicleTypes, Array<FormNode>>([
  [
    VehicleTypes.PSV,
    [
      /*  1 */ TechRecordReasonForCreationSection,
      /*  2 */ PsvNotes,
      /*  3 */ PsvTechRecord,
      /*  4 */ PsvTypeApprovalTemplate,
      /*  5 */ PsvBrakesTemplate,
      /*  6 */ PsvDdaTemplate,
      /*  7 */ DocumentsTemplate,
      /*  8 */ PsvBodyTemplate,
      /*  9 */ PsvWeightsTemplate,
      /* 10 */ PsvTyresTemplate,
      /* 11 */ PsvDimensionsTemplate
    ]
  ],
  [
    VehicleTypes.HGV,
    [
      /*  1 */ TechRecordReasonForCreationSection,
      /*  2 */ NotesTemplate,
      /*  3 */ HgvTechRecord,
      /*  4 */ HgvAndTrlTypeApprovalTemplate,
      /*  5 */ ApplicantDetails,
      /*  6 */ DocumentsTemplate,
      /*  7 */ HgvAndTrlBodyTemplate,
      /*  8 */ HgvWeight,
      /*  9 */ tyresTemplateHgv,
      /* 10 */ HgvDimensionsTemplate,
      /* 11 */ PlatesTemplate
    ]
  ],
  [
    VehicleTypes.TRL,
    [
      /*  1 */ TechRecordReasonForCreationSection,
      /*  2 */ NotesTemplate,
      /*  3 */ TrlTechRecordTemplate,
      /*  4 */ HgvAndTrlTypeApprovalTemplate,
      /*  5 */ ApplicantDetails,
      /*  6 */ DocumentsTemplate,
      /*  7 */ LettersTemplate,
      /*  8 */ HgvAndTrlBodyTemplate,
      /*  9 */ TrlWeight,
      /* 10 */ tyresTemplateTrl,
      /* 11 */ TrlBrakesTemplate,
      /* 12 */ TrlPurchasers,
      /* 13 */ TrlDimensionsTemplate,
      /* 14 */ PlatesTemplate,
      /* 15 */ TrlAuthIntoServiceTemplate,
      /* 16 */ ManufacturerTemplate
    ]
  ],
  [
    VehicleTypes.SMALL_TRL,
    [/* 1 */ TechRecordReasonForCreationSection, /* 2 */ SmallTrailerTechRecord, /* 3 */ ApplicantDetails, /* 4 */ NotesTemplate, /* 5 */ Audit]
  ],
  [
    VehicleTypes.LGV,
    [/* 1 */ TechRecordReasonForCreationSection, /* 2 */ LgvTechRecord, /* 3 */ ApplicantDetails, /* 4 */ NotesTemplate, /* 5 */ Audit]
  ],
  [
    VehicleTypes.CAR,
    [/* 1 */ TechRecordReasonForCreationSection, /* 2 */ CarTechRecord, /* 3 */ ApplicantDetails, /* 4 */ NotesTemplate, /* 5 */ Audit]
  ],
  [
    VehicleTypes.MOTORCYCLE,
    [/* 1 */ TechRecordReasonForCreationSection, /* 2 */ MotorcycleTechRecord, /* 3 */ ApplicantDetails, /* 4 */ NotesTemplate, /* 5 */ Audit]
  ]
]);
