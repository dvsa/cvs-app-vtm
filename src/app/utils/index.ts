export * from './mockStore';
export * from './test-model.utils';
export * from './testing.utils';
export * from './form.utils';


