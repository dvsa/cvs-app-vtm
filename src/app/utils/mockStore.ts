import { map } from 'rxjs/operators';
import { BehaviorSubject } from 'rxjs';
import { Action } from '@ngrx/store';
import { getErrors } from '@app/store/selectors/error.selectors';
import {
  getSelectedVehicleTechRecord,
  getTechViewState,
  getVehicleTechRecordMetaData,
  getActiveVehicleTechRecord,
  selectVehicleTechRecordModelHavingStatusAll
} from '@app/store/selectors/VehicleTechRecordModel.selectors';
import {
  getSelectedVehicleTestResultModel,
  getTestViewState,
  getVehicleTestResultModel,
  selectTestTypeById
} from '@app/store/selectors/VehicleTestResultModel.selectors';
import { getPreparers, getTestStations } from '@app/store/selectors/ReferenceData.selectors';
import { getRouterParams } from '@app/store/selectors/route.selectors';

export class MockStore {
  mockSelector: BehaviorSubject<any>;

  constructor(mockSelector) {
    this.mockSelector = mockSelector;
  }

  select(selector: any) {
    switch (selector) {
      case selectVehicleTechRecordModelHavingStatusAll:
        return this.mockSelector.pipe(
          map((value: any) =>
            value && value.hasOwnProperty('selectVehicleTechRecordModelHavingStatusAll')
              ? value['selectVehicleTechRecordModelHavingStatusAll']
              : []
          )
        );

      case getSelectedVehicleTechRecord:
        return this.mockSelector.pipe(
          map((value: any) =>
            value && value.hasOwnProperty('getSelectedVehicleTechRecord')
              ? value['getSelectedVehicleTechRecord']
              : {}
          )
        );

      case getActiveVehicleTechRecord:
        return this.mockSelector.pipe(
          map((value: any) =>
            value && value.hasOwnProperty('getActiveVehicleTechRecord')
              ? value['getActiveVehicleTechRecord']
              : () => {}
          )
        );

      case getVehicleTechRecordMetaData:
        return this.mockSelector.pipe(
          map((value: any) =>
            value && value.hasOwnProperty('getVehicleTechRecordMetaData')
              ? value['getVehicleTechRecordMetaData']
              : {}
          )
        );

      case getTechViewState:
        return this.mockSelector.pipe(
          map((value: any) =>
            value && value.hasOwnProperty('getTechViewState') ? value['getTechViewState'] : {}
          )
        );

      case getErrors:
        return this.mockSelector.pipe(
          map((value: any) =>
            value && value.hasOwnProperty('getErrors') ? value['getErrors'] : []
          )
        );

      case getVehicleTestResultModel:
        return this.mockSelector.pipe(
          map((value: any) =>
            value && value.hasOwnProperty('getVehicleTestResultModel')
              ? value['getVehicleTestResultModel']
              : []
          )
        );

      case selectTestTypeById(''):
        return this.mockSelector.pipe(
          map((value: any) =>
            value && value.hasOwnProperty('selectTestTypeById') ? value['selectTestTypeById'] : {}
          )
        );

      case getPreparers:
        return this.mockSelector.pipe(
          map((value: any) =>
            value && value.hasOwnProperty('getPreparers') ? value['getPreparers'] : null
          )
        );

      case getTestStations:
        return this.mockSelector.pipe(
          map((value: any) =>
            value && value.hasOwnProperty('getTestStations') ? value['getTestStations'] : null
          )
        );

      case getTestViewState:
        return this.mockSelector.pipe(
          map((value: any) =>
            value && value.hasOwnProperty('getTestViewState') ? value['getTestViewState'] : {}
          )
        );

      case getSelectedVehicleTestResultModel:
        return this.mockSelector.pipe(
          map((value: any) =>
            value && value.hasOwnProperty('getSelectedVehicleTestResultModel')
              ? value['getSelectedVehicleTestResultModel']
              : {}
          )
        );

      case getRouterParams:
        return this.mockSelector.pipe(
          map((value: any) =>
            value && value.hasOwnProperty('getRouterParams') ? value['getRouterParams'] : {}
          )
        );

      default:
        return this.mockSelector;
    }
  }

  dispatch(action: Action) {
    return [];
  }
}
