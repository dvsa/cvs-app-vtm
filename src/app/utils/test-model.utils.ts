import { TestResultModel } from '@app/models/test-result.model';
import { TestType } from '@app/models/test.type';

export const TEST_MODEL_UTILS = {
  mockTestRecord,
  mockTestType
};

function mockTestRecord(args?: Partial<TestResultModel>): TestResultModel {
  const mock: TestResultModel = {
    reasonForCreation: '',
    testResultId: '123',
    vehicleType: 'hgv',
    systemNumber: '123',
    testTypes: [this.mockTestType()],
    testEndTimestamp: '2020-01-16T12:24:38.027Z',
    testStartTimestamp: '2020-01-16T10:24:38.027Z'
  } as TestResultModel;

  return { ...mock, ...args };
}

function mockTestType(args?: Partial<TestType>): TestType {
  const mock: TestType = {
    testTypeId: '1',
    seatbeltInstallationCheckDate: true,
    testTypeName: 'first test',
    testNumber: '123123',
    testTypeEndTimestamp: '2020-01-16T12:24:38.027Z',
    testExpiryDate: '2021-01-16T12:24:38.027Z',
    modType: {
      code: '1',
      description: 'test'
    }
  } as TestType;

  return { ...mock, ...args };
}
