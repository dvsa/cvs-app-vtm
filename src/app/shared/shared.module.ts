import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { DocumentRetrievalService } from '@api/document-retrieval';
import { RoleRequiredDirective } from '@directives/app-role-required.directive';
import { AccordionControlComponent } from './components/accordion-control/accordion-control.component';
import { AccordionComponent } from './components/accordion/accordion.component';
import { BannerComponent } from './components/banner/banner.component';
import { BaseDialogComponent } from './components/base-dialog/base-dialog.component';
import { ButtonGroupComponent } from './components/button-group/button-group.component';
import { ButtonComponent } from './components/button/button.component';
import { IconComponent } from './components/icon/icon.component';
import { NumberPlateComponent } from './components/number-plate/number-plate.component';
import { PaginationComponent } from './components/pagination/pagination.component';
import { TagComponent } from './components/tag/tag.component';
import { TestCertificateComponent } from './components/test-certificate/test-certificate.component';
import { RetrieveDocumentDirective } from './directives/retrieve-document/retrieve-document.directive';
import { PreventDoubleClickDirective } from './directives/prevent-double-click/prevent-double-click.directive';
import { DefaultNullOrEmpty } from './pipes/default-null-or-empty/default-null-or-empty.pipe';
import { DigitGroupSeparatorPipe } from './pipes/digit-group-separator/digit-group-separator.pipe';
import { RefDataDecodePipe } from './pipes/ref-data-decode/ref-data-decode.pipe';
import { TestTypeNamePipe } from './pipes/test-type-name/test-type-name.pipe';

@NgModule({
  declarations: [
    DefaultNullOrEmpty,
    ButtonGroupComponent,
    ButtonComponent,
    BannerComponent,
    RoleRequiredDirective,
    TagComponent,
    NumberPlateComponent,
    IconComponent,
    TestTypeNamePipe,
    AccordionComponent,
    AccordionControlComponent,
    PaginationComponent,
    TestCertificateComponent,
    PreventDoubleClickDirective,
    BaseDialogComponent,
    DigitGroupSeparatorPipe,
    RefDataDecodePipe,
    RetrieveDocumentDirective
  ],
  imports: [CommonModule, RouterModule],
  exports: [
    DefaultNullOrEmpty,
    ButtonGroupComponent,
    ButtonComponent,
    BannerComponent,
    RoleRequiredDirective,
    TagComponent,
    NumberPlateComponent,
    IconComponent,
    TestTypeNamePipe,
    AccordionComponent,
    AccordionControlComponent,
    PaginationComponent,
    TestCertificateComponent,
    BaseDialogComponent,
    DigitGroupSeparatorPipe,
    RefDataDecodePipe,
    RetrieveDocumentDirective
  ],
  providers: [DocumentRetrievalService]
})
export class SharedModule {}
