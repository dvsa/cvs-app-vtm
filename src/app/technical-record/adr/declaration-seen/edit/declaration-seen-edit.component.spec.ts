import { CUSTOM_ELEMENTS_SCHEMA, DebugElement } from '@angular/core';
import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import {
  ReactiveFormsModule,
  FormsModule,
  FormGroupDirective,
  AbstractControl,
  FormGroup
} from '@angular/forms';

import { SharedModule } from '@app/shared';
import { TESTING_UTILS } from '@app/utils/testing.utils';
import { DeclarationSeenEditComponent } from './declaration-seen-edit.component';
import { By } from '@angular/platform-browser';

describe('DeclarationSeenEditComponent', () => {
  let fixture: ComponentFixture<DeclarationSeenEditComponent>;
  let component: DeclarationSeenEditComponent;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [FormsModule, ReactiveFormsModule, SharedModule],
      declarations: [DeclarationSeenEditComponent],
      schemas: [CUSTOM_ELEMENTS_SCHEMA],
      providers: [
        FormGroupDirective,
        {
          provide: FormGroupDirective,
          useValue: TESTING_UTILS.mockFormGroupDirective({
            techRecord: new FormGroup({
              adrDetails: new FormGroup({})
            }) as AbstractControl
          })
        }
      ]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(DeclarationSeenEditComponent);
    component = fixture.componentInstance;
    component.adrDetails = TESTING_UTILS.mockAdrDetails({
      brakeDeclarationsSeen: true,
      brakeEndurance: true,
      weight: '2'
    });
  });

  it('should create with initialized form controls', () => {
    fixture.detectChanges();

    expect(component).toBeDefined();
    expect(component.adrForm.get('brakeDeclarationsSeen').value).toBeTruthy();
    expect(component.adrForm.get('brakeEndurance').value).toBeTruthy();
    expect(component.adrForm.get('weight').value).toEqual('2000');
    expect(fixture).toMatchSnapshot();
  });

  describe('handleFormChanges', () => {
    let ele: DebugElement;

    beforeEach(() => {
      ele = fixture.debugElement;
      component.adrDetails = TESTING_UTILS.mockAdrDetails({
        brakeDeclarationsSeen: true,
        brakeEndurance: true
      });
    });

    it(`should reset brakeDeclarationIssuer & brakeEndurance controls when brakeDeclarationsSeen
    is unchecked`, () => {
      fixture.detectChanges();

      const brakeDeclarationSeen: HTMLInputElement = ele.query(
        By.css('[id=brakeDeclarationsSeen]')
      ).nativeElement;

      brakeDeclarationSeen.click(); // unchecked

      expect(component.adrForm.get('brakeDeclarationIssuer').value).toBeNull();
      expect(component.adrForm.get('brakeEndurance').value).toBeNull();
    });

    describe('brakeEdurance', () => {
      let brakeEdurance: HTMLInputElement;

      beforeEach(() => {
        brakeEdurance = ele.query(By.css('[id=brakeEndurance]')).nativeElement;
      });

      it('should reset the weight control when brakeEndurance is unchecked', () => {
        fixture.detectChanges();

        brakeEdurance.click(); // unchecked
        expect(component.adrForm.get('weight').value).toBeNull();
        expect(component.adrForm.get('weight').validator).toBeNull();
      });

      it('should set required validator when brakeEdurance is checked', () => {
        fixture.detectChanges();

        brakeEdurance.click(); // unchecked
        brakeEdurance.click(); // checked
        expect(component.adrForm.get('weight').validator).toBeDefined();
      });
    });
  });
});
