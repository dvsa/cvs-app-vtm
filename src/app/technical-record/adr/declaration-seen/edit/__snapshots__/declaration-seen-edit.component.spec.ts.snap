// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeclarationSeenEditComponent should create with initialized form controls 1`] = `
<vtm-declaration-seen-edit
  adrDetails={[Function Object]}
  adrForm={[Function FormGroup]}
  detectChange={[Function ViewRef_]}
  fb={[Function FormBuilder]}
  onDestroy$={[Function Subject]}
  parent={[Function FormGroupDirective]}
  removeAdrControl={[Function EventEmitter]}
  validationMapper={[Function ValidationMapper]}
>
  <div
    class="govuk-grid-row"
  >
    <div
      class="govuk-grid-column-full"
    >
      <div
        class="govuk-form-group"
      >
        <fieldset
          class="govuk-fieldset"
        >
          <div
            class="ng-untouched ng-pristine ng-valid"
            ng-reflect-form="[object Object]"
          >
            <h2
              class="govuk-heading-m"
            >
              Declarations seen
            </h2>
            <div
              class="govuk-form-group"
            >
              <div
                class="govuk-checkboxes"
              >
                <div
                  class="govuk-checkboxes__item"
                >
                  <input
                    class="govuk-checkboxes__input ng-untouched ng-pristine ng-valid"
                    formcontrolname="brakeDeclarationsSeen"
                    id="brakeDeclarationsSeen"
                    ng-reflect-name="brakeDeclarationsSeen"
                    type="checkbox"
                    value="true"
                  />
                  <label
                    class="govuk-label govuk-checkboxes__label"
                    for="brakeDeclarationsSeen"
                  >
                     Manufacturer brake declaration 
                  </label>
                </div>
                <div
                  class="govuk-inset-text govuk-!-margin-top-0"
                  id="conditional-brakeDeclarationsSeen"
                >
                  
                  <div
                    class="govuk-form-group govuk-!-margin-left-3"
                  >
                    <label
                      class="govuk-label govuk-!-font-weight-bold"
                      for="brakeDeclarationIssuer"
                    >
                       Issuer 
                    </label>
                    <input
                      class="govuk-input govuk-input--width-10 ng-untouched ng-pristine ng-valid"
                      id="brakeDeclarationIssuer"
                      name="brakeDeclarationIssuer"
                      ng-reflect-name="brakeDeclarationIssuer"
                      type="text"
                    />
                  </div>
                  <div
                    class="govuk-checkboxes__item govuk-!-margin-left-3"
                  >
                    <input
                      class="govuk-checkboxes__input ng-untouched ng-pristine ng-valid"
                      formcontrolname="brakeEndurance"
                      id="brakeEndurance"
                      ng-reflect-name="brakeEndurance"
                      type="checkbox"
                      value="true"
                    />
                    <label
                      class="govuk-label govuk-checkboxes__label"
                      for="brakeEndurance"
                    >
                       Brake endurance 
                    </label>
                  </div>
                  <div
                    class="govuk-inset-text govuk-!-margin-top-0 govuk-!-margin-left-3"
                    id="conditional-brakeEndurance"
                  >
                    
                    <div
                      class="govuk-form-group govuk-!-margin-left-3"
                    >
                      <label
                        class="govuk-label govuk-!-font-weight-bold"
                        for="weight"
                      >
                         Weight(kg) 
                      </label>
                      <input
                        class="govuk-input govuk-input--width-10 ng-untouched ng-pristine ng-valid"
                        id="weight"
                        name="weight"
                        ng-reflect-name="weight"
                        type="text"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="govuk-checkboxes__item"
                >
                  <input
                    class="govuk-checkboxes__input ng-untouched ng-pristine ng-valid"
                    formcontrolname="declarationsSeen"
                    id="declarationsSeen"
                    ng-reflect-name="declarationsSeen"
                    type="checkbox"
                    value="false"
                  />
                  <label
                    class="govuk-label govuk-checkboxes__label"
                    for="declarationsSeen"
                  >
                     Owner/operator declaration 
                  </label>
                </div>
              </div>
            </div>
            
          </div>
        </fieldset>
      </div>
    </div>
  </div>
</vtm-declaration-seen-edit>
`;
